js/reference.js

```diff
@@ -828,10 +837,14 @@ var ERMrest = (function(module) {
         },
 
         get appLink() {
-            if (this._context)
-                return this._table._getAppLink(this._context);
-            else
-                return undefined; // no context
+            if (this._context) {
+                var tag = this._table._getAppLink(this._context);
+                if (tag && module._appLinkFn) {
+                    return module._appLinkFn(tag, this._location);
+                }
+            }
+
+            return undefined; // reference has no context or app link not specified by annotation
```

Comment:
I think we should use the "default" context (if it's available) even if the reference is not contextualized i.e., `this._context` is undefined; instead of returning undefined.

If the default context was not available then we can return undefined.